dev-client/src/components/home/MapSearch.tsx

```diff
@@ -42,14 +42,46 @@ export default function MapSearch({zoomTo, zoomToUser, toggleMapLayer}: Props) {
   const {t} = useTranslation();
   const [query, setQuery] = useState('');
   const [suggestions, setSuggestions] = useState<Suggestion[]>([]);
+  const [_abortController, setAbortController] =
+    useState<AbortController | null>(null);
   //const [isLoading, setIsLoading] = useState(false);
   const [hideResults, setHideResults] = useState(false);
 
-  async function querySuggestions() {
-    if (query.length > 2 || suggestions.length !== 0) {
-      const {suggestions: newSuggestions} = await getSuggestions(query);
+  async function makeSuggestionsApiCall(queryText: string) {
+    const newAbortController = new AbortController();
+    setAbortController(current => {
+      if (current !== null) {
+        current.abort();
+      }
+      return newAbortController;
+    });
+    const newSuggestions = await getSuggestions(
+      queryText,
+      newAbortController.signal,
+    );
+    setAbortController(null);
+    return newSuggestions;
```

Comment:
comment: this took me a second to wrap my head around, i thought there might be a bug where when a request gets cancelled, it sets the next request's abort controller to `null`, because i always forget if you `await` a request that fails the method throws. i think this retry logic should be somehow abstracted and encapsulated somewhere else rather than living in a component's implementation, but that seems tricky, could you file a follow-up tech debt issue for that?

Reply:
Yeah, will do.

Reply:
https://github.com/techmatters/terraso-mobile-client/issues/388